### PR TITLE
Merge zlib, curl and libxml2 into a single job

### LINKED ISSUE #825: Merge zlib, curl and libxml2 into a single job
Right now, `zlib`, `curl` and `libxml2` are built as separate parallel jobs. However, the jobs depending on the build artifacts from these are already waiting on the `compilers` job to complete, making the parallelism pointless. These 3 jobs should be merged and run as a single job, tentatively called `target-libs`.

### PR DESCRIPTION
**Pull request by Steelskin**
These 3 jobs are all configured the same way and do not need to be separate jobs each spinning a new build bot instance.

Fixes #825

### CODE REVIEW

File: .github/workflows/swift-toolchain.yml [hunk 1]
```diff
@@ -969,23 +969,17 @@ jobs:
           symbolsFolder: ${{ github.workspace }}/BinaryCache/1
           searchPattern: '**/*.exe'
 
-  zlib:
+  target_libs:
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:972]:
```suggestion
  dependencies:
```

File: .github/workflows/swift-toolchain.yml [hunk 2]
```diff
@@ -969,23 +969,17 @@ jobs:
...
     runs-on: ${{ inputs.default_build_runner }}
 
     strategy:
       fail-fast: false
       matrix: ${{ fromJSON(inputs.target_matrix) }}
 
-    name: ${{ matrix.os }} ${{ matrix.arch }} zlib
+    name: ${{ matrix.os }} ${{ matrix.arch }} Target libs
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:979]:
```suggestion
    name: ${{ matrix.os }} ${{ matrix.arch }} Dependencies
```

File: .github/workflows/swift-toolchain.yml [hunk 3]
```diff
@@ -1010,14 +1004,22 @@ jobs:
         uses: hendrikmuhs/ccache-action@2e0e89e8d74340a03f75d58d02aae4c5ee1b15c6
         with:
           max-size: 100M
-          key: ${{ steps.workspace_hash.outputs.hash }}-${{ matrix.os }}-${{ matrix.arch }}-zlib
+          key: ${{ steps.workspace_hash.outputs.hash }}-${{ matrix.os }}-${{ matrix.arch }}-target-libs
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:1007]:
```suggestion
          key: ${{ steps.workspace_hash.outputs.hash }}-${{ matrix.os }}-${{ matrix.arch }}-dependencies
```

File: .github/workflows/swift-toolchain.yml [hunk 4]
```diff
@@ -1755,7 +1668,7 @@ jobs:
   sdk:
     # TODO: Build this on macOS or make an equivalent Mac-only job
     if: inputs.build_os == 'Windows'
-    needs: [libxml2, curl, zlib, compilers, cmark_gfm, stdlib, macros]
+    needs: [target_libs, compilers, cmark_gfm, stdlib, macros]
```
Comment (compnerd) [inline .github/workflows/swift-toolchain.yml:1671]:
```suggestion
    needs: [dependencies, compilers, cmark_gfm, stdlib, macros]
```

**Comment by compnerd** [comment]
I'm not sure if we want to necessarily merge these into a single job, there is some value in the parallelism still. I was thinking more that we want to merge the Darwin and Windows builds into a single instance rather than a separate set for Darwin and a separate set for Windows.

**Comment by Steelskin** [comment]
> I'm not sure if we want to necessarily merge these into a single job, there is some value in the parallelism still. I was thinking more that we want to merge the Darwin and Windows builds into a single instance rather than a separate set for Darwin and a separate set for Windows.

Ah you would like to re-merge the flows instead? That's something that should be doable after the mac work is finished. I think this is still a good change because this is only consumed by the sdk job, which is happening after the compilers job so we don't get much from parallelization and this should be a bit cheaper to run since we don't need to spawn new instances for every sub-job.

**Comment by compnerd** [comment]
I am unsure about the merge because I'm thinking that we may want to consume zlib and libxml2 in the compilers build as well, but not CURL.

**Pull request closed without merging**